src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -27251,6 +27264,128 @@ static void M1<T>(T t1)
             comp2.VerifyDiagnostics(expected);
         }
 
+        [Fact]
+        public void AllowNull_Property_WithNotNull()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class COpen<TOpen>
+{
+    [AllowNull, NotNull]public TOpen P1 { get; set; } = default!;
```

Comment:
why use `default!` here? (it feels like AllowNull should prevent any warning here)